gemonlinedb/include/gem/onlinedb/SystemTopology.h

```diff
@@ -0,0 +1,136 @@
+#ifndef GEM_ONLINEDB_SYSTEMTOPOLOGY_H
+#define GEM_ONLINEDB_SYSTEMTOPOLOGY_H
+
+#include <array>
+#include <vector>
+
+#include "gem/onlinedb/AMC13Configuration.h"
+#include "gem/onlinedb/AMCConfiguration.h"
+#include "gem/onlinedb/OHv3Configuration.h"
+#include "gem/onlinedb/VFAT3ChipConfiguration.h"
+
+#include "gem/onlinedb/ConfigurationTraits.h"
+#include "gem/onlinedb/PartReference.h"
+
+namespace gem {
+    namespace onlinedb {
+
+        /**
+         * @brief Describes the topology of a system.
+         *
+         * The description is built as a tree of nodes containing part
+         * references for the various components. The root node is the AMC 13
+         * (only one AMC 13 is supported), under which AMCs are found, then OHs
+         * and VFATs.
+         *
+         * The tree can be read from an XML document that adheres to the
+         * @c system-topology schema.
+         */
+        class SystemTopology
+        {
+        public:
+            /**
+             * @brief Represents a VFAT 3.
+             */
+            class VFAT3Node
+            {
+                friend class SystemTopology;
+
+            public:
+                /**
+                 * @brief The part reference associated to the VFAT.
+                 * @note The @c PartType for @ref VFAT3ChannelConfiguration
+                 *       isn't used.
+                 */
+                ConfigurationTraits<VFAT3ChipConfiguration>::PartType reference;
+
+            private:
+                /// @brief Populates a @c VFAT3Node from XML data.
+                void populate(const xercesc::DOMElement *el);
+            };
+
+            /**
+             * @brief Represents an OHv3.
+             */
+            class OHv3Node
+            {
+                static const constexpr std::size_t VFAT_COUNT = 24;
```

Comment:
I would pass the number of VFATs as a parameter, so that the code can easily accommodate GE2/1

Reply:
See discussion [here](https://github.com/cms-gem-daq-project/cmsgemos/pull/231#discussion_r242139340). TL;DR: will be changed to constants from another header, so that a build-time switch between GE1/1 and GE2/1 can be achieved easily.